drivers/hmis/app/graphql/mutations/create_draft_form.rb

```diff
@@ -0,0 +1,44 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  # This mutation is only for creating a new draft of a published form.
+  # To create a brand-new form, use CreateFormDefinition
+  class CreateDraftForm < CleanBaseMutation
+    argument :identifier, String, required: true
+
+    field :form_identifier, Types::Forms::FormIdentifier, null: true
+
+    def resolve(identifier:)
+      raise 'not allowed' unless current_user.can_manage_forms?
+
+      definitions = Hmis::Form::Definition.
+        exclude_definition_from_select.
+        where(identifier: identifier)
+
+      raise 'not found' if definitions.empty?
+
+      existing_drafts = definitions.where(status: Hmis::Form::Definition::DRAFT)
+      raise 'not allowed to create draft if one already exists' unless existing_drafts.empty?
```

Comment:
This seems fine, another approach would be to just return the draft that is already present. The scenario would occur if a different user created a draft while you had the page open, so the page lets you hit "new draft" even though the draft already exists in the db. We haven't really tackled the question of multiple editors yet- I created https://github.com/open-path/Green-River/issues/6214 so we could discuss it separately.